Deprecate boot2docker-cli officially

tianon opened the pull request:
This is a WIP and not ready for merge.  **DO NOT MERGE YET**.  This is being opened so we can discuss how to orchestrate and communicate the official deprecation properly, and how we'll manage the release of 1.8 to make sure users have a way to continue using this old code if they absolutely need to while still encouraging them very strongly to move off of it ASAP.

cc @ehazlett @nathanleclaire

tianon commented:
An example of how this looks:
```console
$ ./boot2docker shellinit

  WARNING: 'boot2docker' is officially deprecated.
    Please switch to 'docker-machine' (https://github.com/docker/machine)
    as soon as possible.

error in run: Failed to get machine "boot2docker-vm": machine does not exist (Did you run `boot2docker init`?)
```

nathanleclaire commented:
Yeah, that looks pretty good.  I think it might be more useful to link them to https://docs.docker.com/machine/ though.  Would at least save them a click.

ehazlett commented:
Looks good -- +1 for the link to the docs as that tends to be our recommended entry point.

tianon commented:
Cool, that one's an easy fix. :+1:

tianon commented:
Updated the `upgrade` code to only download releases that match `1.x` or `v1.x`, and removed the "Docker client upgrade" portion as that's likely to only cause pain in the future.

nathanleclaire commented on cmds.go:
```diff
@@ -422,46 +411,6 @@ func upgradeBoot2DockerBinary() error {
 	return nil
 }
 
-func upgradeDockerClientBinary() error {
```
I'll never forget you, upgrade code 

/me _sheds single tear_

tianon replied:
![](http://img1.wikia.nocookie.net/__cb20130420051931/starwars/images/4/49/FuneralPyre.png)

SvenDowideit commented:
LGTM - can we also add a proposed EOL? (deprecation only starts a clock)

bfirsh commented:
It might be worth recommending that users download Toolbox to install Machine. Nearly everyone using the boot2docker CLI will have got it from the installers, so Toolbox is the replacement from their POV.

/cc @moxiegirl @mchiang0610

ehazlett commented:
SGTM

On Mon, Aug 10, 2015 at 2:03 PM, Ben Firshman <notifications@github.com>
wrote:

> It might be worth recommending that users download Toolbox to install
> Machine. Nearly everyone using the boot2docker CLI will have got it from
> the installers, so Toolbox is the replacement from their POV.
>
> /cc @moxiegirl <https://github.com/moxiegirl> @mchiang0610
> <https://github.com/mchiang0610>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/boot2docker/boot2docker-cli/pull/377#issuecomment-129551210>
> .
>

mchiang0610 commented:
@bfirsh +1.  Should I create a separate PR for this?

tianon commented:
I'm not a big fan of the HTTP endpoint logic, but it's all in now.  I also backed out the "> v1.x" exclusion, and I've merged Nathan's PR that includes the updated language mentioning Toolbox.

tianon commented:
@SvenDowideit I think an explicit EOL date will convey that we're actually still supporting this up until that point, which is a disservice to our users IMO.  We'll still "support" it, but it's going to get even less attention than it was before, so I think we're essentially saying that it's EOL _today_ barring major or severe issues.

tianon commented:
![image](https://cloud.githubusercontent.com/assets/161631/9212740/7b2662b2-4040-11e5-8e98-489652b5db91.png)

A screenshot of a chat totally counts as a LGTM, right? :smile:

nathanleclaire commented:
LOL u devious

cblage commented:
is it possible to create a light boot2docker wrapper for the docker toolbox?
to facilitate migrating :D

cblage commented:
like a boot2docker-cli-wrapper-docker-toolbox

cblage commented:
alias boot2docker="..."

nathanleclaire commented:
@cblage You might be able to set a bash function which "converts" b2d-cli commands to Docker Machine (e.g. `boot2docker up` becomes `docker-machine start default`) but it will likely be fragile and could potentially fall over in a variety of places.
